packages/web-components/fast-element/src/components/element-controller.ts

```diff
@@ -45,6 +45,7 @@ export class ElementController<TElement extends HTMLElement = HTMLElement>
     private hasExistingShadowRoot = false;
     private _template: ElementViewTemplate<TElement> | null = null;
     private _isConnected: boolean = false;
+    private connecting = false;
```

Comment:
Should we instead have a single variable that tracks the lifecycle, perhaps using some bit flags? We could sill keep `isConnected` as a getter that checks the flags.

Reply:
I made connectedness quad-state but didn't expose it as public, leaving only `isConnected` public. Do you this we should *also* expose this new property publicly and deprecate `isConnected`?

Reply:
I do think there is value in making this public. But, we probably want an easy way to checked connected because there are a lot of those sorts of checks in our foundation component classes. What about this:

* Change the property to `stage` on `ElementController` and make it public.
* Remove the `isConnected` property on `ElementController`
* Add a new `$controllerConnected` property on `FASTElement` itself
* Update foundation classes to use `this.$controllerConnected` so they don't have to do `this.$controller.stage === ControllerStage.connected`

Just trying to think through how we could get some balance of information via the API combined with ergonomics for this common case.

Also note, if we change this, we need to fix up "respectConnection" code in DI.

Reply:
Maybe we should take this to an issue an discuss further? I'm in favor of what you're proposing, but I also think there may still be a gap that isn't filled. Many components care about controller 'connectedness' because they want to know if they can access properties populated by template rendering, and those two things aren't really the same.